From 0a0d6fdd0c6626fa6d7bf16b53f67c94bf298a88 Mon Sep 17 00:00:00 2001
From: Matthew Waters <matthew@centricular.com>
Date: Wed, 14 Feb 2018 23:22:16 +1100
Subject: [PATCH] Fix build with android NDK r16

getifaddrs() may only be available if the target API is >= 24
---
 configure.ac | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 5fabdb4..081b00e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -88,8 +88,9 @@ AH_VERBATIM([_FORTIFY_SOURCE],
 AC_DEFINE([NICEAPI_EXPORT], [ ], [Public library function implementation])
 AC_CHECK_HEADERS([arpa/inet.h net/in.h netdb.h])
 AC_CHECK_HEADERS([ifaddrs.h], \
-		      [AC_DEFINE(HAVE_GETIFADDRS, [1], \
-		       [Whether getifaddrs() is available on the system])])
+		      [AC_CHECK_FUNCS([getifaddrs], \
+ 		          [AC_DEFINE(HAVE_GETIFADDRS, [1], \
+		           [Whether getifaddrs() is available on the system])])])
 AC_CHECK_TYPES([size_t, ssize_t])
 
 # Also put matching version in LIBNICE_CFLAGS
-- 
2.16.1

